odl/solvers/advanced/proximal_operators.py

```diff
@@ -124,14 +133,180 @@ def cconj_prox_factory(step_size):
         Returns
         -------
         proximal : `Operator`
-            The proximal operator of ``prox[a * f^*](x)``
+            The proximal operator of ``s * F^*`` where ``s`` is the step size
         """
-        prox_other = step_size * proximal(1.0 / step_size) * (1.0 / step_size)
+        prox_other = (step_size * prox_factory(1.0 / step_size) *
+                      (1.0 / step_size))
         return IdentityOperator(prox_other.domain) - prox_other
 
     return cconj_prox_factory
 
 
+def proximal_translation(prox_factory, y):
+    """Calculate the proximal of the translated function F(x - y).
+
+    This is calculated according to the rule
+
+        prox[s * F( . - y)](x) = y + prox[s * F](x - y)
+
+    where ``a`` is the translation, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    y : Element in domain of F
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [ComPes2011]_.
+    """
+
+    def translation_prox_factory(step_size):
+        """Create proximal for the translation with a given step_size.
+
+        Parameters
+        ----------
+        step_size : positive `float`
+            Step size parameter
+
+        Returns
+        -------
+        proximal : `Operator`
+            The proximal operator of ``s * F( . - y)`` where ``s`` is the
+            step size
+        """
+
+        return (ConstantOperator(y) + prox_factory(step_size) *
+                (IdentityOperator(y.space) - ConstantOperator(y)))
+
+    return translation_prox_factory
+
+
+def proximal_arg_scaling(prox_factory, a):
+    """Calculate the proximal of function F(x * 1/a).
+
+    This is calculated according to the rule
+
+        prox[s * F( . * 1/a)](x) = a * prox[s/a^2 * F ](x * 1/a)
+
+    where ``a`` scaling parameter, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    a : non-zero `float`
+        Scaling parameter
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [ComPes2011]_.
+    """
+
+    if a == 0:
```

Comment:
If you change definition, wouldn't you get 1/a in front of the prox in the formula? Because if so then you still need to check ```if a == 0```.

However, with this new definition then a = 0 would return the Identity operator (right?). So you could do the check in the beginning and simply return this it is true. Is this better?